uui/hooks/useUUIError.ts

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react';
+import { useUuiContext } from '../services';
+import { ApiCallInfo, ApiRecoveryReason, UuiError, UuiErrorInfo } from '../types';
+
+export type UuiErrorType = 'error' | 'notification' | 'recovery';
+
+export type UUIRecoveryErrorInfo = {
+    title: string,
+    text: string,
+};
+
+export const useUUIError = (customErrorHandler: (error: Error | UuiError | ApiCallInfo, defaultErrorInfo: UuiErrorInfo) => UuiErrorInfo, options: { error: Record<string, UuiErrorInfo>, notification: string, recovery: Record<ApiRecoveryReason, UUIRecoveryErrorInfo> }) => {
+    const [, updateState] = React.useState<object>();
+    const forceUpdate = React.useCallback(() => updateState({}), []);
+    const [errorType, setErrorType] = useState<UuiErrorType>();
+    const [errorInfo, setErrorInfo] = useState<UuiErrorInfo | ApiCallInfo[]>();
+    const { uuiApi, uuiErrors, uuiRouter } = useUuiContext();
+    const { error, recovery, notification } = options;
+
+    useEffect(() => {
+        uuiApi.subscribe(onApiChange);
+        uuiRouter.listen(onRouteChange);
+    }, []);
+
+    const onApiChange = () => {
+        const apiErrors: ApiCallInfo[] = [];
```

Comment:
remove subscription on api change here. Add reporting api errors to the ErrorContext from ApiContext.